sys/net/link_layer/ng_netdev_eth/ng_netdev_eth.c

```diff
@@ -415,7 +415,7 @@ static int _marshall_ethernet(ng_netdev_eth_t *dev, uint8_t *buffer, ng_pktsnip_
     if (netif_hdr->flags & NG_NETIF_HDR_FLAGS_BROADCAST) {
         _addr_set_broadcast(hdr->dst);
     }
-    else if (netif_hdr->flags & NG_NETIF_HDR_FLAGS_MULTICAST) {
+    else if ((netif_hdr->flags & NG_NETIF_HDR_FLAGS_MULTICAST) && (payload != NULL)) {
```

Comment:
Sorry, that wasn't intended. ;) Are you okay with the way it is done in the squash commit?